openscap_report/scap_results_parser/parsers/rule_parser.py

```diff
@@ -20,10 +48,16 @@ def __init__(self, root, test_results, ref_values):
 
     @staticmethod
     def _get_references(rule):
+        url_to_ref_ids = collections.defaultdict(list)
+        for reference_el in rule.findall(".//xccdf:reference", NAMESPACES):
+            url = reference_el.get("href")
+            ref_id = reference_el.text
+            url_to_ref_ids[url].append(ref_id)
         references = []
-        for referenc in rule.findall(".//xccdf:reference", NAMESPACES):
-            references.append(Reference(referenc.get("href"), referenc.text))
-        return references
+        for url, ref_ids in url_to_ref_ids.items():
+            name = KNOWN_REFERENCES.get(url, url)
```

Comment:
I would suggest creating a default value for the `name` if the `href` attribute is an empty string.
For example, reference `SRG-OS-000003-VMM-000030` of the `xccdf_org.ssgproject.content_rule_account_disable_post_pw_expiration` rule.

Reply:
Nice catch!  You're right, the `href` attribute of the `reference` element is optional.

I have add code that accounts for this situation.

However, this situation shouldn't happen in our content. The `reference` without `href` makes little sense. Do you have a specific SCAP content or its version where this happens?

Reply:
I checked the latest content and all the href attributes are not empty strings. 

I found this in:
SSG: `[0, 1, 63]`
Rule:  `xccdf_org.ssgproject.content_rule_account_disable_post_pw_expiration`
Reference: `SRG-OS-000003-VMM-000030`